118 frontend   rework edituserprofile page to allow mutliple location entry

ReginaFera13 opened the pull request:
Changed form to handle new inputs for country, state and city
Reworked setting the location logic to allow user to set multiple locations
Created a way for users to remove a location
Created functionality for auto-populated suggestions as the user types
Refactored rendering of location on Profile Page to handle new data structure and render in correct format

https://github.com/crystaljobe/change-mate/assets/141180011/71b229d1-bc4b-4594-959c-2d7a6cee153f

crystaljobe commented:
Hey Shawn, looks good to me . Just two minor requests - 

Since we have a lot of useStates now it's getting hard to remember which one does what, so would you just mind to add  comments the ones you added or changed to specify what each state is for or when to use it. That way no one uses the wrong states in the wrong way. 

The other thing is the rendering of the multi locations on the profile page card component is looking a little wonky, not a high priority or anything but if you don't want to work on that right now would you mind to just add an issue under the "backlog" iteration. 

Thanks!

ReginaFera13 commented:
Added comments to useStates
Fixed spacing issue on User Profile Location Display

![profile-page-location-format](https://github.com/crystaljobe/change-mate/assets/141180011/946423fb-dfdf-482d-b910-1d7d597f35ef)

crystaljobe commented:
Beautiful, thank you for doing that!
